src/main/java/com/sourcery/gymapp/backend/workout/dto/CreateWorkoutExerciseSetDto.java

```diff
@@ -0,0 +1,28 @@
+package com.sourcery.gymapp.backend.workout.dto;
+
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Positive;
+import jakarta.validation.constraints.PositiveOrZero;
+import jakarta.validation.constraints.Size;
+
+import java.math.BigDecimal;
+
+public record CreateWorkoutExerciseSetDto(
+
+        @Positive
+        @NotNull
+        Integer setNumber,
+
+        @Positive
+        Integer reps,
+
+        @PositiveOrZero
+        BigDecimal weight,
```

Comment:
Does @PositiveOrZero means that it also cannot be @NotNull ?

Reply:
It can be null, I think we don't need to validate this field if someone for example wants first to add empty sets as a template -> click save, and then after some time just to fill the data 🤔